### PR TITLE
Changed Sharepoint policy 1.4 to not-implemented since it relies on deprecated field RequireAcceptingAccountMatchInvitedAccount

### LINKED ISSUE #1267: Sharepoint policy 1.4 setting RequireAcceptingAccountMatchInvitedAccount seems to be deprecated or ignored
# 🐛 Summary #

The Sharepoint admin setting for policy 1.4 "Guests must sign in using the same account to which sharing invitations are sent", backend field RequireAcceptingAccountMatchInvitedAccount looks like it is being deprecated or simply ignored. We can no longer locate this setting in any of the test tenant Sharepoint admin centers and when we execute the code below, it does nothing and the value of the variable always comes back false. [We are not the only ones experiencing this problem](https://github.com/microsoft/Microsoft365DSC/issues/4771).

It is not clear yet but we might need to turn off or remove this from Scuba, because it doesn't make sense to have a policy that A) doesn't exist in the admin center and B) the user cannot configure via Powershell either.

```
Set-PnPTenant -RequireAcceptingAccountMatchInvitedAccount $true
(Get-PnPTenant).RequireAcceptingAccountMatchInvitedAccount # this always returns $false
```

image

image

This is the respective setting described in the baseline.

image

### PR DESCRIPTION
**Pull request by tkol2022**
#  #
  

## 🗣 Description ##

Modified the implementation of Sharepoint 1.4 so that it is "not implemented" in the report since we now have evidence that it is deprecated or at least a user cannot configure it anymore.

Closes #1267 

## 🧪 Testing ##

Tested using interactive authentication and service principal against the E5 tenant. Verified that the report produces the expected output message.

image

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] PR targets the correct parent branch (e.g., main or release-name) for merge.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] Changes are sized such that they do not touch excessive number of files.
- [x] *All* future TODOs are captured in issues, which are referenced in code comments.
- [x] These code changes follow the ScubaGear [content style guide](https://github.com/cisagov/ScubaGear/blob/main/CONTENTSTYLEGUIDE.md).
- [x] Related issues these changes resolve are linked preferably via [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [x] All relevant type-of-change labels added.
- [x] All relevant project fields are set.
- [x] All relevant repo and/or project documentation updated to reflect these changes.
- [x] Unit tests added/updated to cover PowerShell and Rego changes.
- [x] Functional tests added/updated to cover PowerShell and Rego changes.
- [x] All relevant functional tests passed.
- [x] All automated checks (e.g., linting, static analysis, unit/smoke tests) passed.

## ✅ Pre-merge checklist ##

- [x] PR passed smoke test check.
- [x] Feature branch has been rebased against changes from parent branch, as needed

  Use `Rebase branch` button below or use [this](https://www.digitalocean.com/community/tutorials/how-to-rebase-and-update-a-pull-request) reference to rebase from the command line.
- [x] Resolved all merge conflicts on branch
- [x] Notified merge coordinator that PR is ready for merge via comment mention

## ✅ Post-merge checklist ##

- [ ] Feature branch deleted after merge to clean up repository.
- [ ] Verified that all checks pass on parent branch (e.g., main or release-name) after merge.

### CODE REVIEW

**Comment by tkol2022** [comment]
Testers please run ScubaGear with both interactive auth and service principal since we can to ensure Sharepoint works as expected with both.

File: PowerShell/ScubaGear/Rego/Utils/ReportDetails.rego [hunk 1]
```diff
@@ -44,6 +44,13 @@ NotCheckedDetails(PolicyId) := sprintf(
     [PolicyLink(PolicyId)]
 )
 
+# Use this when you need to make a policy not-implemented because of deprecation
+NotCheckedDeprecation(PolicyId) :=
+    concat(" ", [
+    "This configuration setting has been deprecated and we are in the process of removing it from the baseline."
+    ])
```
[the remark this answers is not in the document]
Comment (tkol2022) [reply PowerShell/ScubaGear/Rego/Utils/ReportDetails.rego]: thx much for catching this. i updated it to remove the parameter not being used.

**Review by mitchelbaker-cisa — APPROVED** [review]
Tested generating a report with interactive authentication with success; ran functional tests and the SharePoint 1.4 revisions for both spo/pnp passed as expected; smoke tests pass. 

As a future TODO we should create an issue to track deletion of SharePoint 1.4 when the time comes. The task would require modifying [removedpolicies.md](https://github.com/cisagov/ScubaGear/blob/main/PowerShell/ScubaGear/baselines/removedpolicies.md).

**Comment by tkol2022** [comment]
> Tested generating a report with interactive authentication with success; ran functional tests and the SharePoint 1.4 revisions for both spo/pnp passed as expected; smoke tests pass.
> 
> As a future TODO we should create an issue to track deletion of SharePoint 1.4 when the time comes. The task would require modifying [removedpolicies.md](https://github.com/cisagov/ScubaGear/blob/main/PowerShell/ScubaGear/baselines/removedpolicies.md).

@ahuynhMITRE Can u kindly take care of that? thank you much.

**Review by james-garriss — APPROVED** [review]
Ran SPO and PNP tests.  No problems.

**Comment by tkol2022** [comment]
@nanda-katikaneni ready for merge

**Pull request merged**